### ISSUE TITLE
Image stacking does not work with stills

### ISSUE THREAD

**Issue by biochem-fan** [dials/dials#2724]
Image stacking in `dials.image_viewer` does not work for multi-frame containers (e.g. MPCCD from SACLA [dials-data/image_examples/SACLA-MPCCD-Phase3-21528-5images.h5](https://github.com/dials/data-files/blob/master/image_examples/SACLA-MPCCD-Phase3-21528-5images.h5)). This is probably because frames are interpreted as multiple experiments, not a single image set.

Indeed, importing the image with `dials.import convert_stills_to_sequence=True` allows image stacking. But this comes with a side effect that per-shot wavelengths are ignored (#2725).

**Comment by graeme-winter** [comment]
@biochem-fan 

> But this comes with a side effect that per-shot wavelengths are ignored.

If this is an image sequence you can still have one experiment per image, but with them using the same underlying image sequence, with a `scan` to handle the indexing. 

This can be made to work

**Comment by biochem-fan** [comment]
@graeme-winter
I see. Indeed I manually edited the `imported.expt` and confirmed that `dials.show` displayed per-shot wavelengths.
This is worth doing.

But this alone does not solve the problem with `dials.image_viewer XXXX.h5`, where importing happens implicitly.
Potential options are:

1. fix image stacking for multiple experiments
2. make `convert_stills_to_sequence=True` the default

I prefer 2, because this also solves #2447.

**Comment by biochem-fan** [comment]
> 1. fix image stacking for multiple experiments

The patch below implements this. I will send a PR together with a fix for #1512.

```diff
diff --git a/src/dials/util/image_viewer/spotfinder_frame.py b/src/dials/util/image_viewer/spotfinder_frame.py
index 56e7304e1..b46978460 100644
--- a/src/dials/util/image_viewer/spotfinder_frame.py
+++ b/src/dials/util/image_viewer/spotfinder_frame.py
@@ -1087,15 +1087,12 @@ def stack_images(self):
             if not isinstance(image_data, tuple):
                 image_data = (image_data,)
 
-            i_frame = self.image_chooser.GetClientData(
-                self.image_chooser.GetSelection()
-            ).index
-            imageset = self.images.selected.image_set
+            i_frame = self.image_chooser.GetSelection()
 
             for i in range(1, self.params.stack_images):
-                if (i_frame + i) >= len(imageset):
+                if (i_frame + i) >= len(self.images):
                     break
-                image_data_i = imageset[i_frame + i]
+                image_data_i = self.images[i_frame + i].get_image_data()
                 for j, rd in enumerate(image_data):
                     data = image_data_i[j]
                     if mode == "max":
```
